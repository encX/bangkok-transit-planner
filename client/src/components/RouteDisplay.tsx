import React, {FunctionComponent} from 'react';

interface RouteDisplayProps {
	test?: string;
}

export const RouteDisplay: FunctionComponent<RouteDisplayProps> = (props: RouteDisplayProps): JSX.Element => {
	return (
		<div/>
	);
};
