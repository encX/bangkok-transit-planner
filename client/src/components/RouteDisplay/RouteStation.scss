@import "src/css/Common";

.route-station {
	&--time {
		padding-top: $station-dot-pad-top;
		text-align: right;
	}
}
