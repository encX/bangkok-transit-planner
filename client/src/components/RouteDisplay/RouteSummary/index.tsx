import React, {FC} from 'react';
import {RouteSummaryItem} from './RouteSummaryItem';

interface RouteSummaryProps {
	cost: number;
	time: string;
	transfer: number;
}

const RouteSummary: FC<RouteSummaryProps> = ({cost, time, transfer}): JSX.Element => {
	return (
		<>
			<RouteSummaryItem name="Fare" value={`${cost} Baht`}/>
			<RouteSummaryItem name="Time" value={time}/>
			<RouteSummaryItem name="Transfer" value={`${transfer} times`}/>
		</>
	);
};

export default RouteSummary;
