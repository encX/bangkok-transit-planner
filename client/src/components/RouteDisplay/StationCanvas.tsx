import React, {FC} from 'react';

import './StationCanvas.scss';

interface StationCanvasProps {
	mode: 'start' | 'middle' | 'end';
	color?: string;
}

export const StationCanvas: FC<StationCanvasProps> = ({mode, color}): JSX.Element => {
	return (
		<div className="station-canvas">
			<div className={`station-canvas--top station-canvas--top-${mode}`} style={{backgroundColor: color}}/>
			<div className={`station-canvas--bottom station-canvas--bottom-${mode}`} style={{backgroundColor: color}}/>
			<div className="station-canvas--dot" style={{borderColor: color}}/>
		</div>
	);
};
