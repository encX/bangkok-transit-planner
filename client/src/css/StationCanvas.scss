@import "Common";

$default-color: #888;

.station-canvas {
	width: 100%;
	height: 100%;
	display: flex;
	flex-direction: column;
	align-items: center;

	&--top {
		width: $route-line-width;
		height: $station-dot-pad-top + 10;
		&-start { background: none !important }
		&-middle, &-end { background-color: $default-color }
	}

	&--bottom {
		width: $route-line-width;
		flex-grow: 1;
		&-start, &-middle { background-color: $default-color }
		&-end { background: none !important }
	}
	&--dot {
		width: $station-dot-size;
		height: $station-dot-size;
		background-color: #FFF;
		border: $default-color $route-line-width solid;
		border-radius: $station-dot-radius;
		position: absolute;
		top: $station-dot-pad-top;
	}
}
