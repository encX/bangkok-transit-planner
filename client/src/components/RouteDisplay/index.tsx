import React, {FunctionComponent} from 'react';

import {Route} from '../../api/models/Route';
import RouteSummary from './RouteSummary';
import {RouteStation} from './RouteStation';

const RouteDisplay: FunctionComponent<Route> = ({stations, cost}: Route): JSX.Element => {
	const start = stations.slice(0, 1)[0];
	const between = stations.slice(1, -1);
	const end = stations.slice(-1)[0];
	
	return(
		<div className="route-display">
			<RouteSummary cost={cost} time="0 min" transfer={0}/>
			<RouteStation mode="start" stationName={start.name}/>
			{between.map((station, i) => <RouteStation mode="middle" key={i} stationName={station.name}/>)}
			<RouteStation mode="end" stationName={end.name}/>
		</div>
	);
};

export default RouteDisplay;
