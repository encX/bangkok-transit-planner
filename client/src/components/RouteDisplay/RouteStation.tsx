import {Col, Row, Typography} from 'antd';
import React, {FunctionComponent} from 'react';
import {StationCanvas} from './StationCanvas';

import './RouteStation.scss';

const {Title} = Typography;

interface RouteStationProps {
	mode: 'start' | 'middle' | 'end';
	stationName: string;
}

export const RouteStation: FunctionComponent<RouteStationProps> = ({mode, stationName}): JSX.Element => {
	return (
		<Row gutter={[16, 0]}>
			<Col flex="90px"><div className="route-station--time">11:11 AM</div></Col>
			<Col flex="40px"><StationCanvas mode={mode} /></Col>
			<Col flex="auto"><Title>{stationName}</Title></Col>
		</Row>
	);
};
