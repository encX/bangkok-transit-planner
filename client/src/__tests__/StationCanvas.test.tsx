import React from 'react';
import {shallow} from 'enzyme';
import {StationCanvas} from '../components/RouteDisplay/StationCanvas';


describe('StationCanvas', () => {
	it('case "start": renders correctly', () => {
		const tree = shallow(<StationCanvas mode="start" color="#123123"/>);
		expect(tree).toMatchSnapshot();
	});
	
	it('case "middle": renders correctly', () => {
		const tree = shallow(<StationCanvas mode="middle" color="#123123"/>);
		expect(tree).toMatchSnapshot();
	});
	
	it('case "end": renders correctly', () => {
		const tree = shallow(<StationCanvas mode="end" color="#123123"/>);
		expect(tree).toMatchSnapshot();
	});
});
