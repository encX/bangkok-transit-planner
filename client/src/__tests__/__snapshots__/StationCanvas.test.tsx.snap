// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationCanvas case "end": renders correctly 1`] = `
<div
  className="station-canvas"
>
  <div
    className="station-canvas--top station-canvas--top-end"
    style={
      Object {
        "backgroundColor": "#123123",
      }
    }
  />
  <div
    className="station-canvas--bottom station-canvas--bottom-end"
    style={
      Object {
        "backgroundColor": "#123123",
      }
    }
  />
  <div
    className="station-canvas--dot"
    style={
      Object {
        "borderColor": "#123123",
      }
    }
  />
</div>
`;

exports[`StationCanvas case "middle": renders correctly 1`] = `
<div
  className="station-canvas"
>
  <div
    className="station-canvas--top station-canvas--top-middle"
    style={
      Object {
        "backgroundColor": "#123123",
      }
    }
  />
  <div
    className="station-canvas--bottom station-canvas--bottom-middle"
    style={
      Object {
        "backgroundColor": "#123123",
      }
    }
  />
  <div
    className="station-canvas--dot"
    style={
      Object {
        "borderColor": "#123123",
      }
    }
  />
</div>
`;

exports[`StationCanvas case "start": renders correctly 1`] = `
<div
  className="station-canvas"
>
  <div
    className="station-canvas--top station-canvas--top-start"
    style={
      Object {
        "backgroundColor": "#123123",
      }
    }
  />
  <div
    className="station-canvas--bottom station-canvas--bottom-start"
    style={
      Object {
        "backgroundColor": "#123123",
      }
    }
  />
  <div
    className="station-canvas--dot"
    style={
      Object {
        "borderColor": "#123123",
      }
    }
  />
</div>
`;
